docs/rfcs/0000-search-defaults.md

```diff
@@ -0,0 +1,101 @@
+
+## Reference-level explanation
+
+### Region
+
+We will introduce a new (internal) `RegionManager` class that will take care of tracking and updating the _current_ and _home_ region of the user. A `RegionMiddleware` will listen to the `InitAction` of the `BrowserStore` and dispatch an action to add a `RegionState` to the `BrowserState`, as well as asking the `RegionManager` to check whether a region change has happened (by internally querying MLS).
+
+The new `SearchMiddleware` (see [RFC 2 about the new search architecture](https://mozac.org/rfc/0002-search-state-in-browser-store)) will listen to the `RegionState` update, load the matching default search engines using the search storage, and dispatch an action updating the `SearchState`.
+
+## Rationale and alternatives
+
+* An earlier draft of this RFC described using multiple region sources (GNSS, Network and SIM country, Google Play Services) and weighting them. This requires collecting more real world data about the accuracy of those sources and how they drift from what MLS determines, before shipping an implementation and therefore this idea was discarded here. It may be the topic for a separate, future RFC.
+
+## Prior art
+
+* [Timezone check in Firefox desktop](https://searchfox.org/mozilla-central/source/toolkit/modules/Region.jsm#180)
+* [Update interval in Firefox desktop](https://searchfox.org/mozilla-central/source/toolkit/modules/Region.jsm#75)
+* [RFC about moving state to new AC component](https://mozac.org/rfc/0002-search-state-in-browser-store)
+* [Fuzzy location provider idea](https://github.com/mozilla-mobile/android-components/issues/1720)
+* [SearchEngineManager in Fennec](https://searchfox.org/mozilla-esr68/source/mobile/android/base/java/org/mozilla/gecko/search/SearchEngineManager.java)
+* Current [SearchEngineManager in Android Components](https://github.com/mozilla-mobile/android-components/blob/08880314f56d73691b3cd909d5dee199bba4ed0b/components/browser/search/src/main/java/mozilla/components/browser/search/SearchEngineManager.kt#L28)
+* [FenixSearchEngineProvider](https://github.com/mozilla-mobile/fenix/blob/master/app/src/main/java/org/mozilla/fenix/components/searchengine/FenixSearchEngineProvider.kt) in Fenix
+* [SearchService](https://searchfox.org/mozilla-central/source/toolkit/components/search/SearchService.jsm) in Firefox (desktop)
+* [SearchEngines.swift](https://github.com/mozilla-mobile/firefox-ios/blob/main/Client/Frontend/Browser/SearchEngines.swift) in Firefox for iOS
+
+## Unresolved questions
+
+* What kind of telemetry could or should we collect? What data would help us tweak this? How can we confirm accuracy?
+
+* Is the 2 weeks period before switching the _home_ region, that we borrowed from desktop, acceptable for mobile too?
+
+* Would it help to expose the _current_ and _home_ region in Fenix in a debug screen or in the "about" screen?
```

Comment:
It could be helpful for debugging future situations

Reply:
Desktop has exposed the home region at least in about:support and it has been useful there when we've been looking at issues with people having the wrong/unexpected search engines.